src/Playwright.Tests/BaseTests/PlaywrightAssert.cs

```diff
@@ -39,7 +39,7 @@ internal static class PlaywrightAssert
             try
             {
                 await action();
-                Assert.Fail();
+                Assert.Fail($"Expected exception of type '{typeof(T).Name}' which was not thrown");
```

Comment:
this gives a better error message when a exception was not thrown inside an assertion